resource-service/src/main/java/uk/gov/gchq/palisade/service/resource/web/ResourceController.java

```diff
@@ -18,117 +18,70 @@
-import uk.gov.gchq.palisade.service.ConnectionDetail;
-import uk.gov.gchq.palisade.service.ResourceService;
-
-import java.util.Map;
-import java.util.concurrent.ExecutionException;
+import uk.gov.gchq.palisade.service.resource.request.AddResourceRequest;
+import uk.gov.gchq.palisade.service.resource.request.GetResourcesByIdRequest;
+import uk.gov.gchq.palisade.service.resource.request.GetResourcesByResourceRequest;
+import uk.gov.gchq.palisade.service.resource.request.GetResourcesBySerialisedFormatRequest;
+import uk.gov.gchq.palisade.service.resource.request.GetResourcesByTypeRequest;
+import uk.gov.gchq.palisade.service.resource.service.StreamingResourceServiceProxy;
 
 @RestController
 @RequestMapping(path = "/")
 public class ResourceController {
 
     private static final Logger LOGGER = LoggerFactory.getLogger(ResourceController.class);
 
-    private ResourceService service;
+    private final StreamingResourceServiceProxy service;
 
-    public ResourceController(final ResourceService service) {
+    public ResourceController(final StreamingResourceServiceProxy service) {
         this.service = service;
     }
 
-    @PostMapping(path = "/getResourcesById", consumes = "application/json", produces = "application/json")
-    public Map<LeafResource, ConnectionDetail> getResourcesById(@RequestBody final GetResourcesByIdRequest request) {
+    @PostMapping(path = "/getResourcesById", consumes = "application/json", produces = "application/octet-stream")
+    public ResponseEntity<StreamingResponseBody> getResourcesById(@RequestBody final GetResourcesByIdRequest request) {
         LOGGER.info("Invoking getResourceById: {}", request);
-        Map<LeafResource, ConnectionDetail> response = getResourcesByIdRequest(request);
-        LOGGER.info("Returning response {}", response);
-        return response;
+        StreamingResponseBody stream = out -> service.getResourcesById(request.getResourceId(), out);
+        LOGGER.info("Streaming response: {}", stream);
+        return new ResponseEntity<>(stream, HttpStatus.OK);
     }
 
-    @PostMapping(path = "/getResourcesByResource", consumes = "application/json", produces = "application/json")
-    public Map<LeafResource, ConnectionDetail> getResourcesByResource(@RequestBody final GetResourcesByResourceRequest request) {
+    @PostMapping(path = "/getResourcesByResource", consumes = "application/json", produces = "application/octet-stream")
+    public ResponseEntity<StreamingResponseBody> getResourcesByResource(@RequestBody final GetResourcesByResourceRequest request) {
         LOGGER.info("Invoking getResourcesByResource: {}", request);
-        Map<LeafResource, ConnectionDetail> response = getResourceByResourceRequest(request);
-        LOGGER.info("Returning response {}", response);
-        return response;
+        StreamingResponseBody stream = out -> service.getResourcesByResource(request.getResource(), out);
+        LOGGER.info("Streaming response: {}", stream);
```

Comment:
Logging level needs to be debug/trace for some of these, only one log write at info.

Reply:
Agreed, this is kind of redundant anyway as the stream is always just a java.util.Stream..........